trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/shared/component/StatePlaceholder.kt

```diff
@@ -0,0 +1,90 @@
+package net.thechance.mena.trends.presentation.shared.component
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import mena.trends_presentation.generated.resources.Res
+import mena.trends_presentation.generated.resources.ic_no_connection
+import mena.trends_presentation.generated.resources.re_try
+import net.thechance.mena.designsystem.presentation.component.button.PrimaryButton
+import net.thechance.mena.designsystem.presentation.component.text.Text
+import net.thechance.mena.designsystem.presentation.theme.theme.MenaTheme
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+import org.jetbrains.compose.ui.tooling.preview.Preview
+
+@Composable
+fun StatePlaceholder(
+    modifier: Modifier = Modifier,
+    isBottomVisible: Boolean = false,
```

Comment:
I suggest to rename this to : 
```suggestion
    showRetryButton: Boolean = false,
```

Reply:
I’d prefer to keep it as isBottomVisible because this component is meant to be reusable.
It might contain other buttons or elements in the future, not just the retry button 
so naming it `showRetryButton` would make it less generic.

Reply:
if add another buttons or any composable in the future they will have different names and you will need control the visibility of them according to screen , so there are two options  : 
1- pass the bottom section as composable function and every screen will pass it's content to it 
2- add every different component by it's name in parameter like showRetryButton 

It's my opinion :) 
@ahmeddsamii what's your opinion?

Reply:
i agree with @HendSayed25 , and i prefer to go with pass the composable so that every state will use this should implement all the missing parts according to its screen